packages/deployer/internal/ast/storage-validator.js

```diff
@@ -11,32 +17,63 @@ function findDuplicateStorageNamespaces(contracts) {
   }
   const duplicates = findDuplicateSlots(slots);
 
+  const errors = [];
```

Comment:
Following pattern used by router validation: utils return an array of errors found instead of just logging them.